xbmc/addons/AddonInstaller.h

```diff
@@ -173,6 +175,7 @@ class CAddonInstallJob : public CFileOperationJob
   ADDON::RepositoryPtr m_repo;
   bool m_isUpdate;
   bool m_isAutoUpdate;
+  const char* m_currentType = "";
```

Comment:
make this an enum, don't abuse string compares for state describing

Reply:
There you right. 
Just used it as it is in its parent (https://github.com/xbmc/xbmc/blob/master/xbmc/utils/Job.h#L137).

I look then to use a:
```cpp
void CAddonInstaller::OnJobComplete(unsigned int jobID, bool success, CJob* job)
{
  CAddonInstallJob* addonJob = dynamic_cast<CAddonInstallJob*>(job);
```
To get another type of value.

Reply:
Haven't looked at the parent class, then add the following in this file and use those variables.

```
static constexpr char* TYPE_DOWNLOAD = "DOWNLOAD";
static constexpr char* TYPE_INSTALL = "INSTALL";
```